src/Controller/StatusController.php

```diff
@@ -31,6 +41,57 @@ public function statusAction(Request $request)
         return $this->render('@Clifton/Status/status.html.twig', [
             'now' => new DateTime(),
             'dbConnectivity' => $dbalConnection->isConnected() || $dbalConnection->connect(),
+            'dbErrorCheck' => $hasDBIssues,
         ]);
     }
+
+    /**
+     * This insanity is copy/pasted from Faucet. It detects whether an Exception indicates that the
+     * database is down. No, there is no single exception for that.
+     * When the database is down we return a 200 status. Other DB exceptions return a 500 status.
+     * See programmes ticket https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5534
+     */
+    private function testForNonConnectionDatabaseIssues()
+    {
+        try {
+            $pid = new Pid('b006m86d'); //Eastenders
+            $programme = $this->get('pps.programmes_service')->findByPidFull($pid);
```

Comment:
Ooh conflicted. `findByPidFull` joins to a bunch of tables that `findByPid` does not. But 'findByPidFull' is called more often so the DB cache is likely to be warm. I wonder which would be faster to process?